### ISSUE TITLE
Change forecast.io references to darksky

### ISSUE THREAD

**Issue by kaushalmodi** [cadadr/forecast.el#13]
Hi,

I have been happily using this package :)

Today I got this email that forecast.io is now darksky.

Thankfully the forecast.io API still works.

But this package might need to update forecast.io references to darksky.
Example: `darksky-api-key` instead of `forecast-api-key`, `[Ff]orecast.io` references all over the code, etc

**Comment by cadadr** [comment]
Hi,

sorry for the delay, a bit busy these days.  I've seen the mail but I didn't have the chance to read it in depth and see if the API will be the same, though the new URL seems to work (I just updated `forecast-api-url` to `"https://api.darksky.net"`).  If you confirm that it works, changing the default value of the aforementioned variable and updates to documentation would suffice to catch up on the new APIs, so I'll do that.  Re: the variable names, well, I hope to port this thing to something open like openweathermap.org one day, so I do not want its names to refer to a certain service provider (I did not think the name forecast.el as a reference to forecast.io initially, but it seems to be so).

I'm glad that you use this package and that you're fond of it!
